### PR TITLE
Do not allow XR session on non user-activated events

### LINKED ISSUE #23787: Check for user activation in WebXR
`requestSession()` is not supposed to work in immersive mode if it isn't triggered by user activation. We should also update `simulateUserActivation()` from https://github.com/servo/servo/pull/23786 when implementing this.

### PR DESCRIPTION
**Pull request by paulrouget**
This sets the thread in "user interaction mode" when the dispatched event is trusted. I also tried an approach where we would not rely on the dispatched event but just set "user interaction mode" when we get a compositor event (which, we can assume, are only user generated). That worked as well.

Fixes #23787.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24133)

### CODE REVIEW

**Comment by paulrouget** [comment]
Alright. Gonna use the compositor event approach instead.

**Comment by paulrouget** [comment]
Done.

**Comment by CYBAI** [comment]
```
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Requesting immersive session outside of a user gesture rejects", 
    "test": "/webxr/xrDevice_requestSession_immersive_no_gesture.https.html", 
    "line": 374587, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": "Test timed out", 
    "stack": null, 
    "subtest": "create_session", 
    "test": "/_mozilla/webxr/create_session.html", 
    "line": 393534, 
    "action": "test_result", 
    "expected": "PASS"
}
{
    "status": "ERROR", 
    "group": "default", 
    "message": "The operation is insecure.", 
    "stack": "undefined:undefined:undefined", 
    "subtest": null, 
    "test": "/_mozilla/webxr/create_session.html", 
    "line": 393535, 
    "action": "test_result", 
    "expected": "OK"
}
{
    "status": "FAIL", 
    "group": "default", 
    "message": "promise_test: Unhandled rejection with value: object \"SecurityError: The operation is insecure.\"", 
    "stack": null, 
    "subtest": "obtain_frame", 
    "test": "/_mozilla/webxr/obtain_frame.html", 
    "line": 394168, 
    "action": "test_result", 
    "expected": "PASS"
}
{
    "status": "FAIL", 
    "group": "default", 
    "message": "promise_test: Unhandled rejection with value: object \"SecurityError: The operation is insecure.\"", 
    "stack": null, 
    "subtest": "Ensure that XRWebGLLayer's constructor throws appropriate errors", 
    "test": "/webxr/xrWebGLLayer_constructor.https.html", 
    "line": 401177, 
    "action": "test_result", 
    "expected": "TIMEOUT"
}
{
    "status": "OK", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/webxr/xrWebGLLayer_constructor.https.html", 
    "line": 401178, 
    "action": "test_result", 
    "expected": "ERROR"
}
```

**Comment by atouchet** [comment]
This looks like it needs to be restarted.

**Comment by jdm** [comment]
This needs a `./mach fmt`, sadly.

File: tests/wpt/web-platform-tests/webxr/xrWebGLLayer_constructor.https.html [hunk 1]
```diff
@@ -17,8 +17,11 @@
   let gl = webglCanvas.getContext('webgl', glAttributes);
   return navigator.xr.test.simulateDeviceConnection(TRACKED_IMMERSIVE_DEVICE)
     .then(() => {
-      return navigator.xr.requestSession('inline')
-        .then((session) => {
+      let sessionPromise;
+      navigator.xr.test.simulateUserActivation(function() {
+        sessionPromise = navigator.xr.requestSession('inline');
```
Comment (Manishearth) [inline tests/wpt/web-platform-tests/webxr/xrWebGLLayer_constructor.https.html]: Note: this was a mistake, inline sessions with no tracking data (which must be explicitly requested via features) don't need activation

Comment (jdm) [reply tests/wpt/web-platform-tests/webxr/xrWebGLLayer_constructor.https.html]: File an issue? That sounds like a mistake in our implementation.

Comment (Manishearth) [reply tests/wpt/web-platform-tests/webxr/xrWebGLLayer_constructor.https.html]: In other words, we don't have enough of an implementation of the prerequisites to write the code for this. Paul's implementation is correct for the features we have right now, it's just the test change that isn't. Alex Cooper is already fixing the test so we're fine, I just wanted to note this.

**Pull request merged**